src/data_prepration/data_prep.py

```diff
@@ -0,0 +1,142 @@
+import math
+import os
+import re
+import cv2
+from pathlib import Path
+import matplotlib.pyplot as plt
+import matplotlib.image as mpimg
+import numpy as np
+import av
+
+def display_image(input_dir) :
+    """
+    Display an image.
+    :param input_dir: Path to directory containing image.
+    """
+    test_images_input = os.listdir(input_dir)
+
+    stopper=0
+    for img in test_images_input:
+        print(img)
+        stopper+=1
+        image = mpimg.imread( os.path.join(input_dir,img))
+        print('Shape of image: ',image.shape)
+        plt.imshow(image)
+        plt.show()
+        if stopper==1:
+            break
+    pass
+
+def rgb_2_single(input_dir, output_dir):
+    """Applies the Grayscale transform
+    This will return an image with only one color channel but NOTE: to see the returned image as grayscale
+    (assuming your grayscaled image is called 'gray') you should call plt.imshow(gray, cmap='gray')
+    """
+
+    test_images_input = os.listdir(input_dir)
+    if not os.path.isdir(output_dir):
+        os.makedirs(output_dir)
+
+    for img in test_images_input:
+        gray_image = cv2.imread(os.path.join(input_dir,img), cv2.IMREAD_GRAYSCALE)
+        base_filename= os.path.basename(img)
+        cv2.imwrite(output_dir+'/'+os.path.basename(img)[:-9]+'gray'+'.png', gray_image)
+        counter+=1
+    pass
+
+
+def create_videos(frames_path):
```

Comment:
For creating a video from sequence of images, FFmpeg maybe easier than code in opencv:
```bash
ffmpeg -i frame_%d.png -c:v libx264 -vf fps=25 mv.mp4
```